src/Google.Api.Gax/ResourceNameList.cs

```diff
@@ -52,6 +52,18 @@ public ResourceNameList(IList<string> underlyingList, Func<string, TName> string
         /// <inheritdoc />
         public void Add(TName item) => _underlyingList.Add(_nameToString(item));
 
+        /// <summary>
+        /// Adds all items to this list.
+        /// </summary>
+        /// <param name="items">The items to add to this list.</param>
+        public void Add(IEnumerable<TName> items)
+        {
+            foreach (TName item in GaxPreconditions.CheckNotNull(items, nameof(items)))
```

Comment:
I'd probably move the nullity check to a separate line here, but I don't mind much.